pkg/plugins/policies/meshaccesslog/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,150 @@
+package v1alpha1
+
+import (
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshAccessLogResource) validate() error {
+	var verr validators.ValidationError
+	spec := validators.RootedAt("spec")
+
+	r.validateTop(spec, &verr)
+	r.validateFrom(spec, &verr)
+	r.validateTo(spec, &verr)
+	r.validateIncompatibleCombinations(spec, &verr)
+	r.validateToOrFromDefined(spec, &verr)
```

Comment:
So what came from the boilerplate generator wasn't good enough?

Reply:
If not it would be nice to improve it maybe?

Reply:
The boilerplate generator generated some of the checks (like which targets are valid for which `targetRef`) but it did not check other things. I guess we could have some structured comments / annotations on the `proto` that would be read by the generator and generate code validations but:
1. Not sure it's a scope of this PR.
2. Is it worth investing time in it?
3. Does anything like that already exists?

Reply:
Can you not add the extra validation after the code generated by the boilerplate? My guess is that if the validators always read the same way it will just be quicker and more efficient to read the code.

Reply:
> I guess we could have some structured comments / annotations on the proto that would be read by the generator and generate code validations

People have tried this in the past, validation usually just needs to be written as code because it's complex.

Reply:
> Can you not add the extra validation after the code generated by the boilerplate?

This is exactly what it is just wrapped into functions, e.g. [this part](https://github.com/kumahq/kuma/pull/4998/files#diff-514fa6f8fc6be662c82f1dcb020fbf431c014218a289efe0645deae0664f5b2eR87-R98) was generated.

> My guess is that if the validators always read the same way it will just be quicker and more efficient to read the code.

To me it's easier to read 5 functions in the `validate()` method and roughly know what's going on here, rather than have one function with 120 lines and have to read the whole thing to understand what's going on. 

If you prefer the parts generated to be untouched I suggest having those parts wrapped in a function and a comment suggesting not to edit those parts.

Reply:
we can adjust the boilerplate generator to generate a code like:
```go
func (r *MeshAccessLogResource) validate() error {
	var verr validators.ValidationError
	spec := validators.RootedAt("spec")

	r.validateTop(spec, &verr)
	r.validateFrom(spec, &verr)
	r.validateTo(spec, &verr)
}
```

Reply:
Yes I think @lobkovilya 's suggestion makes sense.

Reply:
[Issue](https://github.com/kumahq/kuma/issues/5005) created.